bindings/serialize.h

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SERIALIZE
+#define SERIALIZE
+
+#include <memory>
+#include <map>
+#include "v8-profiler.h"
+
+std::vector<char> serializeHeapProfile(std::unique_ptr<v8::AllocationProfile>
+  profile, int64_t intervalBytes, int64_t startTimeNanos);
+
+std::vector<char> serializeTimeProfile(std::unique_ptr<v8::CpuProfile>,
+  int64_t samplingIntervalMicros, int64_t startTimeNanos);
+
+class ProtoMessage {
+public:
+  virtual void encode(std::vector<char> &b) = 0;
+};
+
+class ValueType : public ProtoMessage {
+public:
+  int64_t typeX;
+  int64_t unitX;
+  ValueType(int64_t typeX=0, int64_t unitX=0);
```

Comment:
I think spaces around = are required per clang-format style but we really need to get the whole source clang-format'ed before further comments on style are meaningful.

Reply:
done.